enterprise/ha/src/main/java/org/neo4j/kernel/ha/cluster/SwitchToSlave.java

```diff
@@ -432,8 +447,19 @@ private void catchUpWithMaster() throws IllegalArgumentException, InterruptedExc
         userLog.info( "Catching up with master. I'm at %s", catchUpRequestContext );
 
         // Unpause the update puller, because we know that we are a slave that just started communication with master.
-        updatePuller.unpause();
-        updatePuller.await( UpdatePuller.NEXT_TICKET, true );
+        while ( true )
```

Comment:
This loop has no guarantee that it will terminate. We've had issues with similar constructs lately and i am a bit sceptical as to their safety. I'd prefer either logging failures with exponential back off on retries, or a bound number of iterations and raising a failure if not successful.